algorithm/dinic.h

```diff
@@ -30,52 +31,67 @@ namespace whfc {
 		int direction = 0;
 		std::vector<PinIndex> current_flow_sending_pin, current_flow_receiving_pin, current_pin;
 		std::vector<InHeIndex> current_hyperedge;
-		
+
 		Flow upperFlowBound = std::numeric_limits<Flow>::max();
-		
-		DinicBase(FlowHypergraph& hg) : hg(hg), queue(hg.numNodes()), stack(hg.numNodes()),
+
+		DinicBase(FlowHypergraph& hg) : hg(hg), num_nodes(hg.numNodes()), num_edges(hg.numHyperedges()),
+                    queue(hg.numNodes()), stack(hg.numNodes()),
 										current_flow_sending_pin(hg.numHyperedges(), PinIndex::Invalid()),
 										current_flow_receiving_pin(hg.numHyperedges(), PinIndex::Invalid()),
 										current_pin(hg.numHyperedges(), PinIndex::Invalid()),
 										current_hyperedge(hg.numNodes(), InHeIndex::Invalid())
 		{
-		
+
 		}
-		
+
 		void flipViewDirection() {
 			std::swap(current_flow_sending_pin, current_flow_receiving_pin);
 			direction = 1 - direction;
 		}
-		
+
 	};
-	
+
 	class Dinic : public DinicBase {
 	public:
 		using Type = Dinic;
-		
+
 		static constexpr bool same_traversal_as_grow_assimilated = false;
 		static constexpr bool grow_reachable_marks_flow_sending_pins_when_marking_all_pins = true;
 		static constexpr bool log = false;
-		
+
 		Dinic(FlowHypergraph& hg) : DinicBase(hg)
 		{
 			reset();
 		}
-		
+
 		void reset() {
-		
+      if ( hg.numNodes() > num_nodes ) {
+        num_nodes = hg.numNodes();
+        LayeredQueue<Node> tmp_queue(num_nodes);
+        std::swap(queue, tmp_queue);
```

Comment:
use resize

Reply:
actually, we should just remove the pre-alloc from the queue and stack...I'll do that later